### PR TITLE
Simplifying colors

### PR DESCRIPTION
**Pull request by JohnOberhauser**
@devotaaabel I think you were just copying what firefox did for their theme colors, right?  It seems a little complex to me, and I'm trying to figure out if there is any benefits to it.  When I was adding some colors to the other day, I found it tedious having to update the constructor, the variables, the update method, and the copy method.  Any reason we can't simplify it like so?

### CODE REVIEW

**Review by timc-mozilla — APPROVED** [review]
looks cleaner :D

File: core/designsystem/src/main/java/org/mozilla/social/core/designsystem/theme/MoSoTheme.kt [hunk 1]
```diff
@@ -89,6 +85,7 @@ private val darkColorPalette = MoSoColors(
     iconAccent = FirefoxColor.Violet30,
     iconActionActive = FirefoxColor.Violet20,
     iconActionDisabled = FirefoxColor.LightGrey70,
+    iconAccentViolet = FirefoxColor.Violet30,
```
Comment (devotaaabel) [inline core/designsystem/src/main/java/org/mozilla/social/core/designsystem/theme/MoSoTheme.kt:88]: is this supposed to have "violet" in the name? Is it from the design system? I didn't see it in the figma though I could've been looking in the wrong place. Seems weird that it has the word "violet" in the name, as no other design system tokens do that.

**Review by devotaaabel — COMMENTED** [review]
@JohnOberhauser re: your question about why- this is how material themes do it- curious if you looked into that/why they do it this way?

File: core/designsystem/src/main/java/org/mozilla/social/core/designsystem/theme/MoSoTheme.kt (hunk 1 above)
Comment (JohnOberhauser) [reply core/designsystem/src/main/java/org/mozilla/social/core/designsystem/theme/MoSoTheme.kt]: I'm not sure.  This was previously on line 245.  I'm not sure how it ended up there.  I removed it and replaced it's usage.

Comment (devotaaabel) [reply core/designsystem/src/main/java/org/mozilla/social/core/designsystem/theme/MoSoTheme.kt]: Ohh I see now, I'm pretty sure it was a typo

**Review by devotaaabel — APPROVED** [review]
Cool thanks for looking into it

**Pull request merged**